src/platform_impl/linux/wayland/event_loop/mod.rs

```diff
@@ -225,35 +222,29 @@ impl<T: 'static> EventLoop<T> {
 
     pub fn pump_events<F>(&mut self, timeout: Option<Duration>, mut callback: F) -> PumpStatus
     where
-        F: FnMut(Event<T>, &RootEventLoopWindowTarget<T>, &mut ControlFlow),
+        F: FnMut(Event<T>, &RootEventLoopWindowTarget<T>),
     {
         if !self.loop_running {
             self.loop_running = true;
 
             // Reset the internal state for the loop as we start running to
             // ensure consistent behaviour in case the loop runs and exits more
             // than once.
-            self.control_flow = ControlFlow::Poll;
+            self.set_control_flow(ControlFlow::Poll);
 
             // Run the initial loop iteration.
             self.single_iteration(&mut callback, StartCause::Init);
         }
 
         // Consider the possibility that the `StartCause::Init` iteration could
         // request to Exit.
-        if !matches!(self.control_flow, ControlFlow::ExitWithCode(_)) {
+        if !self.is_exit() {
             self.poll_events_with_timeout(timeout, &mut callback);
         }
-        if let ControlFlow::ExitWithCode(code) = self.control_flow {
+        if let Some(code) = self.exit_code() {
             self.loop_running = false;
 
-            let mut dummy = self.control_flow;
-            sticky_exit_callback(
-                Event::LoopExiting,
-                self.window_target(),
-                &mut dummy,
-                &mut callback,
-            );
+            callback(Event::LoopExiting, self.window_target());
```

Comment:
This `with_code` and `exit_code` is weird given that we basically return an error and can just forward error, I think the `ExitWithCode` should just be entirely removed out of it and the error must be propagated via the means of `Result`. There's an issue on how users should get their error out of the event loop, but they can clearly just store to external variable as of now.

Reply:
This has already been done like you described. `ControlFlow::ExitWithCode` was removed, `with_code()` and `exit_code()` is internal stuff to some backends that where using `ControlFlow::ExitWithCode` before to pass down non-user error codes.

Cc #3059.

Reply:
I've added an additional notice to the changelog about the removal of `ExitWithCode`.

Reply:
Hm, I guess I just got confused why we even use code, when we can just forward error, like the code arises from the error simply by using `1`, but we already have an error in place which we can bubble up?

Reply:
Yeah, it covers it.